internal/installation/paths.go

```diff
@@ -28,7 +25,20 @@ func DefaultInstallPath() (string, error) {
 	return InstallPathForBranch(constants.BranchName)
 }
 
-func InstallPath() (string, error) {
+func InstallRoot(path string) (string, error) {
+	currentPath := path
+	for filepath.Dir(currentPath) != currentPath {
+		candidate := filepath.Join(currentPath, InstallDirMarker)
+		if fileutils.TargetExists(candidate) {
+			return filepath.Dir(candidate), nil
+		}
+		currentPath = filepath.Dir(currentPath)
+	}
```

Comment:
Can we use [this](https://github.com/ActiveState/cli/blob/master/internal/fileutils/fileutils.go#L389) instead?

Reply:
I'd forgotten about that, good thinking!